docs/config/intro.md

```diff
@@ -0,0 +1,59 @@
+Configuration Files
+===================
+
+Forge has built in support for creating and managing mod configuration files. The purpose of a configuration file is to add flexibility and allow end users to customize their experience.
+
+Creating
+--------
+
+Create a dedicated class. Use the builder instance to define the configuration structure and data.
+
+```java
+private BooleanValue mySetting;
+
+public ServerConfig(ForgeConfigSpec.Builder builder) {
+  builder.comment("examplemod settings"); // Comment at the start of the file
+  
+  builder.push("Example Category"); // 'Enter' the Category
+  
+  mySetting = builder
+    .comment("Set this to true to enable something") // Comment on the setting
+    .translation("examplemod.exampleboolserver") // Language translation key
+    .define("exampleboolserver", false); // Setting key and default value
+
+  builder.pop(); // 'Exit' the category
+}
+```
+
+Registration
+------------
+
+Create the Pair instance.
+
+```java
+protected final static Pair<ServerConfig, ForgeConfigSpec> serverConfigPair = new ForgeConfigSpec.Builder().configure(ServerConfig::new);
+```
+
+```java
+ModLoadingContext.get().registerConfig(ModConfig.Type.SERVER, serverConfigPair.getRight());
```

Comment:
It should be explained what the various config types do and where the configuration files are created in the game folder for each type.